### DISCUSSION TITLE
Remove documentation for StyleLayer?

### DISCUSSION THREAD

**Discussion by drwestco** [maplibre/maplibre-gl-js#3046] [General]
https://maplibre.org/maplibre-gl-js/docs/API/classes/maplibregl.StyleLayer/

StyleLayer is mostly (or entirely) an internal class. The event methods (on/off/once) probably shouldn't be documented, as the supported way to attach/remove an event to a layer is by going through the Map on/off/once methods.

**Comment by HarelM** [comment]
Feel free to submit a PR, I think you are right.
